File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        
        priority_queue<pair<int,char>>pq;
        map<char,int>mp;
        
      for(int i=0;i<s.length();i++)
      {
          mp[s[i]]++;
         
      }
        
        for(auto it:mp)
        {
            pq.push({it.second,it.first});
        }
        string ans;
        while(!pq.empty())
        {
           int p=pq.top().first;
            cout<<p<<" ";
            
            
            while(p--)
            {
                ans+=pq.top().second;
            }
            pq.pop();
            
        }
      
        return ans;
    }
};

File: 0451-sort-characters-by-frequency/0451-sort-characters-by-frequency.py
class Solution:
    def frequencySort(self, s: str) -> str:
        
        max_heap=[]
        
        mp1={}
        
        for ch in s:
            if ch in mp1:
                mp1[ch]+=1
            else:
                mp1[ch]=1
                
        for key,value in mp1.items():
            
            heapq.heappush(max_heap,(-value,key))
            
        ans=""
        
        while max_heap:
            
            freq,char=heapq.heappop(max_heap)
            
            ans+=char*(-freq)
            
        return ans    